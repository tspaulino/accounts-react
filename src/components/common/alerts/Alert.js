import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Message, Icon } from 'semantic-ui-react'

class Alert extends Component {
  componentWillMount() {
    const { revokeAlert, id } = this.props

    // Schedule alerts to be
    // removed after 7 seconds
    this.timer = setTimeout(() => revokeAlert(id), 7000)
  }

  shouldComponentUpdate() {
    return false
  }

  componentWillUnmount() {
    clearTimeout(this.timer)
  }

  render() {
    const { message, type } = this.props
    const props = { [type]: true }
    return (
      <Message {...props}>
        <Icon name="warning" />
        {message}
      </Message>
    )
  }
}

Alert.propTypes = {
  revokeAlert: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
}

export default Alert
