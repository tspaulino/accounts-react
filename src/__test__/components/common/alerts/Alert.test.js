import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'

import { Message, Icon } from 'semantic-ui-react'
import Alert from '../../../../components/common/alerts/Alert'

const props = {
  revokeAlert: sinon.spy(),
  id: 'id',
  type: 'error',
  message: 'Some error'
}

describe('<Alert />', () => {
  it('should render component', () => {
    const wrapper = shallow(<Alert {...props} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(Message).prop('error')).toBeTruthy()
    expect(wrapper.find(Icon).prop('name')).toEqual('warning')
    expect(wrapper.find(Message).render().text()).toEqual('Some error')

    // Make sure the component doesn't update
    wrapper.setProps({ type: 'success' })
    expect(wrapper.find(Message).prop('error')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
