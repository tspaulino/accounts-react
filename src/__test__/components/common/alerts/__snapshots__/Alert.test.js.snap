// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render component 1`] = `
<Message
  error={true}
>
  <Icon
    as="i"
    name="warning"
  />
  Some error
</Message>
`;

exports[`<Alert /> should render component 2`] = `
<Message
  error={true}
>
  <Icon
    as="i"
    name="warning"
  />
  Some error
</Message>
`;
